### ISSUE TITLE
Generated documentation website visual layout (navigation)

### ISSUE THREAD

**Issue by flip111** [purescript/spago#1054]
On the generated documentation website there is this `GROUP BY PACKAGE` checkbox. It's a bit funny UI.

What about instead of a single `Index` in the navigation bar (top right), there will be two items instead `Modules` and `Packages`?

**Comment by f-f** [comment]
I moved this issue into Spago, as this is part of the Halogen app that we add on top of the generated docs. Sources is [here](https://github.com/purescript/spago/tree/master/docs-search/client-halogen) and PRs are most welcome 🙂

**Comment by f-f** [comment]
I don't have an opinion on the proposed change as of now, I think I'd need to see it visualised

**Comment by flip111** [comment]
![image](https://github.com/purescript/spago/assets/2244480/a372ad96-afb3-433e-9783-7e06b8b9de2b)

**Comment by CharlesTaylor7** [comment]
I agree the layout is a bit funky. For now it could be adapted to just be the list of modules. The grouping is not actually by package, but just into 3 groups: "<builtins>", "<local package>" and "everything else". I don't find it super useful.

 
<img width="268" alt="Screen Shot 2023-10-08 at 4 53 12 PM" src="https://github.com/purescript/spago/assets/16541866/433504db-a39f-4163-854a-8af0181cc675">

**Comment by f-f** [comment]
IIRC previously it would actually group by package?
I think having it always grouped by package would be the most useful - sometimes I look for the package, sometimes I look for the module. We should load the page with the list of packages, and expand to the list of modules when clicking on an arrow next to it. Maybe have an "expand all" button too.

**Comment by flip111** [comment]
> We should load the page with the list of packages, and expand to the list of modules when clicking on an arrow next to it. Maybe have an "expand all" button too.

Great idea :+1:

**Comment by f-f** [comment]
This is very close to what the previous Spago/docs-search did. I think the "group by package" feature broke in the migration, because it was more useful then - this picture is generated from the `aff` repo with Spago 0.21.0 and PureScript 0.14.9:

<img width="547" alt="image" src="https://github.com/purescript/spago/assets/5480361/57e86978-4fa4-4140-a8c7-f9ceb8f5e7b5">

We'd keep a similar vibe, but with an arrow next to the package that lets you expand to the list of modules. Right now that heppens when you click on a package, but we should instead redirect to the readme of the package, just like in pursuit.

We'd also take away the checkbox for grouping by package because we'd always group by package like this, and add an "Expand/contract all" button to expand all the module lists in the packages.

**Comment by f-f** [comment]
> Are we likely to have local copies of readmes?

We are. This is not yet a thing (not sure if we even have an issue for this), but we would like to enforce the presence of a README for Registry packages, so that we can embed them in the archives and Pursuit doesn't have to fetch them from somewhere else, but can access them offline.

Adding a "package page" would get us one step closer to "Pursuit in PureScript" as well 🙂
